### PR TITLE
Add basic typescript api

### PR DESCRIPTION
**Pull request by ian-antking**

### CODE REVIEW

File: api/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import app from "./app"
+
+const PORT: string = process.env.PORT || "4000"
```
Comment (C-gyorfi) [inline api/src/index.ts:3]: I wonder if we need to pick a different port, the e2e test docker container runs on port 4000

File: api/src/routes/status.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { Router } from "express"
+import { checkStatus } from "../controllers/status"
+
+const router: Router = Router()
+
+router.get("/", checkStatus)
```
Comment (C-gyorfi) [inline api/src/routes/status.ts:6]: Can we add a test for this endpoint?

**Review by C-gyorfi — APPROVED** [review]
Approving with a few comments

File: api/src/routes/status.ts (hunk 2 above)
Comment (ian-antking) [reply api/src/routes/status.ts:6]: added a basic integration test with supertest

**Review by C-gyorfi — APPROVED** [review]

File: api/src/app.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import * as express from "express"
+import * as cors from "cors"
+import statusRouter from "./routes/status"
+
+const app: express.Application = express()
+app.use(cors())
+
+app.use("/status", statusRouter)
```
Comment (emadkaramad) [inline api/src/app.ts:8]: If we rename it to `/health` we can then use it as a health endpoint for our ecs services.

**Review by emadkaramad — APPROVED** [review]

**Review by emadkaramad — DISMISSED** [review]

File: .eslintrc.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 module.exports = {
   root: true,
   env: { es6: true },
-  ignorePatterns: ["build/*", "jest.setup.ts"],
+  ignorePatterns: ["build/*", "api/build/*", "jest.setup.ts"],
```
Comment (bjpirt) [inline .eslintrc.js:4]: Wondering whether we should just completely ignore everything in the api folder and give it its own config for ts, eslint, etc?

**Review by bjpirt — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 4 above)
Comment (C-gyorfi) [reply .eslintrc.js:4]: I think it would be good to have one linting for the whole repo, assuming we'd use the same setting in both projects

**Review by emadkaramad — DISMISSED** [review]

**Review by bjpirt — DISMISSED** [review]

**Review by C-gyorfi — APPROVED** [review]

**Pull request merged**